include/fastdds/topic/DataWriter.hpp

```diff
@@ -0,0 +1,331 @@
+class PublisherImpl;
+class Publisher;
+
+/**
+ * Class DataWriter, contains the actual implementation of the behaviour of the DataWriter.
+ * @ingroup FASTRTPS_MODULE
+ */
+class DataWriter
+{
+    friend class PublisherImpl;
+
+    /**
+     * Create a data writer, assigning its pointer to the associated writer.
+     * Don't use directly, create Publisher using DomainRTPSParticipant static function.
+     */
+    DataWriter(
+            PublisherImpl* p,
+            TypeSupport type,
+            const fastrtps::TopicAttributes& topic_att,
+            const fastrtps::rtps::WriterAttributes& att,
+            const fastrtps::WriterQos& qos,
+            const fastrtps::rtps::MemoryManagementPolicy_t memory_policy,
+            DataWriterListener* listener = nullptr);
+
+public:
+
+    virtual ~DataWriter();
+
+    /**
+     * Write data to the topic.
+     * @param data Pointer to the data
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data);
+
+    /**
+     * Write data with params to the topic.
+     * @param data Pointer to the data
+     * @param params Extra write parameters.
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data,
+            fastrtps::rtps::WriteParams& params);
+
+    /**
+     * Write data with handle.
+     * @param data Pointer to the data
+     * @param handle InstanceHandle_t.
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data,
+            const fastrtps::rtps::InstanceHandle_t& handle);
+
+    /**
+     *
+     * @param kind
+     * @param  data
+     * @return
+     */
+    bool create_new_change(
+            fastrtps::rtps::ChangeKind_t kind,
+            void* data);
+
+    /**
+     *
+     * @param kind
+     * @param  data
+     * @param wparams
+     * @return
+     */
+    bool create_new_change_with_params(
+            fastrtps::rtps::ChangeKind_t kind,
+            void* data,
+            fastrtps::rtps::WriteParams& wparams);
+
+    /**
+     *
+     * @param kind
+     * @param  data
+     * @param wparams
+     * @param handle
+     * @return
+     */
+    bool create_new_change_with_params(
+            fastrtps::rtps::ChangeKind_t kind,
+            void* data,
+            fastrtps::rtps::WriteParams& wparams,
+            const fastrtps::rtps::InstanceHandle_t& handle);
+
+    /**
+     * Removes the cache change with the minimum sequence number
+     * @return True if correct.
+     */
+    bool remove_min_seq_change();
+
+    /**
+     * Removes all changes from the History.
+     * @param[out] removed Number of removed elements
+     * @return True if correct.
+     */
+    bool remove_all_change(
+            size_t* removed);
```

Comment:
Moved to private section.

Reply:
In fact, do we need `remove_all_change()` and `remove_min_seq_change()`?

Reply:
I kept them because FastRTPS API has these methods (in camel case). They seem to be unused too. Probably all these methods can be safely removed, but I would prefer to ask before.

Reply:
Removed get_remote[writer|reader]_info methods.
removeAllChange is used by a BlackBox test, so I guess it will be useful when the tests use DDS API.